waterfall/golang/server/server.go

```diff
@@ -103,8 +103,9 @@ func (s *WaterfallServer) Push(rpc waterfall_grpc_pb.Waterfall_PushServer) error
 	// the contents of the stream into the desired path
 	eg := &errgroup.Group{}
 	eg.Go(func() error {
-		defer r.Close()
-		return stream.Untar(r, xfer.Path)
+		err := stream.Untar(r, xfer.Path)
+		defer r.CloseWithError(err)
```

Comment:
It's was not clear to me what the behavior is if you call CloseWithError(nil). It turns out a normal Close() does exactly that.